File: Easy-Array/SecondMaxout.java

import java.util.Scanner;

// Write a program to find the Second largest element in the array
class SecondMaxout
{
  public static int SecondMax(int arr[])
  {
    // Initialize maximum element
      int max=-1;
      // Traverse array elements
      for(int i=0;i<arr.length;i++)
      {
        // Check if current element is greater than max
        if(arr[i]>max)
        {
            // Update max
            max=arr[i];
        }
      }
      // Initialize second maximum element
      int secondmax=-1;
      // Traverse array elements
      for(int i=0;i<arr.length;i++)
      {
        // Check if current element is greater than secondmax and less than max
        if(arr[i]>secondmax && arr[i]!=max)
        {
            // Update secondmax
            secondmax=arr[i];
        }
      }
      return secondmax;
  }  
  public static void main(String[] args)
  {
       Scanner sc=new Scanner(System.in);
       System.out.println("Size of Array");
       int n=sc.nextInt();
       int arr[]=new int[n];
       System.out.println("Enter the elements of array");
       for(int i=0;i<n;i++)
       {
        arr[i]=sc.nextInt();
       }
       System.out.println("SecondMaximum element of array is "+SecondMax(arr)); 
  }  
}

File: Easy-Array/SecondMaxout.py
# find the second largest in the array

def secondlargest(arr):
    if not arr:
        return None  # Return None for an empty array

    largest = arr[0]  # Initialize largest with the first element

    for num in arr:
        if num > largest:
            largest = num  # Update largest if current number is greater
    
    secondLargest = -1
    for num in arr:
        if num != largest and num > secondLargest:
              secondLargest=num
    
    return secondLargest

number_of_elements = int(input("Enter the number of elements in the array: "))
array = []
print("Enter the elements of the array:")
for _ in range(number_of_elements):
    element = int(input())
    array.append(element)
largest_element = find_largest(array)
if largest_element is not None:
    print(f"The largest element in the array is: {largest_element}")
else:
    print("The array is empty.")